model/src/datetime/display.rs

```diff
@@ -93,36 +88,25 @@ impl TimestampIso8601Display {
 
 impl Display for TimestampIso8601Display {
     fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
-        /// Approximate number of seconds before we're too far in the future
-        /// to care.
-        pub const APPROXIMATE_MAX_SECONDS: u64 =
-            (YEAR_MAX as u64 - EPOCH_UNIX as u64) * DAYS_PER_YEAR as u64 * SECONDS_PER_DAY;
-
-        let total_seconds = self.timestamp.as_secs();
-        let microseconds = self.timestamp.as_micros() % MICROSECONDS_PER_SECOND;
-
-        if total_seconds >= APPROXIMATE_MAX_SECONDS {
-            return Err(FmtError);
-        }
-
-        let seconds = total_seconds % SECONDS_PER_DAY;
-
-        let Date { day, month, year } = date(total_seconds);
-
         // Years.
         //
         // Input: 2021-01-01T01:01:01.010000+00:00
         //       |----|
+        let year = self.timestamp.0.year();
+
```

Comment:
Does `time`s built in formatter differ from ours or why don't we use it?

Reply:
I didn't use it because for our use case we only need to format as ISO8601, and `time`'s formatter has a lot more complexity and is behind a feature flag.